src/veh_interact.cpp

```diff
@@ -1353,27 +1353,56 @@ void veh_interact::do_refill()
 
 void veh_interact::calc_overview()
 {
-    const hotkey_queue &hotkeys = hotkey_queue::alphabets();
+    overview_opts.clear();
+    overview_headers.clear();
 
-    const auto next_hotkey = [&]( const vehicle_part & pt, input_event & evt ) {
-        if( overview_action && overview_enable && overview_enable( pt ) ) {
-            const input_event prev = evt;
-            evt = main_context.next_unassigned_hotkey( hotkeys, evt );
-            return prev;
-        } else {
-            return input_event();
+    veh_interact::calc_vehicle_header( veh );
+
+    //call for main vehicle ui
+    overview_opts = get_vehicle_overview( veh, false );
+
+    //group carried vehicle parts by their assigned carry_names stacks top element
+    std::map<std::string, std::vector<vehicle_part>> carried_vehicles;
+
+    for( const vpart_reference &vpr : veh->get_all_parts() ) {
+        if( vpr.part().has_flag( vehicle_part::carried_flag ) ) {
+            carried_vehicles[vpr.part().carry_names.top().substr( vehicle_part::name_offset )].push_back(
+                vpr.part() );
         }
-    };
+    }
 
-    overview_opts.clear();
-    overview_headers.clear();
+    for( std::pair<std::string, std::vector<vehicle_part>> it : carried_vehicles ) {
+        vehicle *tmp_car = new vehicle();
```

Comment:
This needs to be freed.
Why are you allocating this on the heap (manually) anyways?
If it's too big to fit on the stack, you can use something like a unique_ptr for automatic memory management.

Reply:
i am not sure i understand.
tmp_car is created inside the loop because i am generating new dummy vehicles for every found carried car to send to the overview generating method.
The stack only contains the names of carried cars, not the cars themselfs.
I am not sure what you mean with freeing the car in this situation? It is enclosed inside the for loop and should be catched by garbage collection once that loop finishes.

Reply:
There is no garbage collection in C++, the `new` keyword allocates memory that must be manually freed later via `delete`.
So you need to either stick this variable on the stack (just do `vehicle tmp_car`), or if you cannot do that for some reason (it is too large), you should use something that automatically does the memory management for you (`std::unique_ptr` being on example).

Reply:
Okay, I wasn't quite clear, you should either:
1. Replace this variable with `vehicle tmp_car;`
2. Or, replace it with `std::unique_ptr<vehicle> tmp_car = std::make_unique<vehicle()>;`, iff the above does not work.

You generally shouldn't use new and delete outside of specific classes internally managing that memory in C++, and in this repo in particular, you pretty much never need to use it.

Also, side note:
Memory allocated with `new` must be freed with `delete`, and memory allocated with `malloc()` must be freed with `free()`.

Reply:
thank you for you help with this issue. it should be fixed now.